### PR TITLE
linux: qspi: Edit OSPI/QSPI Document

### PR DESCRIPTION
**Pull request by santhosh21**
Remove the NOTE for known issue and add some allignment changes.

### CODE REVIEW

File: source/linux/Foundational_Components/Kernel/Kernel_Drivers/QSPI.rst [hunk 1]
```diff
@@ -17,117 +17,89 @@ controllers work only in master mode.
...
+   +---------------+--------------+-------------------------------------------+
 
 .. note::
 
-   Not all OSPI flashes can be supported. Users are recommended to check
-   whether or not the OSPI flash part chosen for custom board designs meets all
-   the criteria listed at https://e2e.ti.com/support/processors/f/791/t/946418
+    Not all OSPI flashes can be supported. Users are recommended to check
+    whether or not the OSPI flash part chosen for custom board designs meets all
+    the criteria listed at https://e2e.ti.com/support/processors/f/791/t/946418
```
Comment (StaticRocket) [inline source/linux/Foundational_Components/Kernel/Kernel_Drivers/QSPI.rst:102]: Whitespace regression

File: source/linux/Foundational_Components/Kernel/Kernel_Drivers/QSPI.rst [hunk 2]
```diff
@@ -210,50 +182,50 @@ Start the Linux Kernel Configuration tool:
 
 .. code-block:: console
 
-        $ make menuconfig ARCH=<architecture_name>
+    $ make menuconfig ARCH=<architecture_name>
```
Comment (StaticRocket) [inline source/linux/Foundational_Components/Kernel/Kernel_Drivers/QSPI.rst:185]: Over 50% of this file is using the proper 3 space indentation. Please standardize these code-blocks if you are going to adjust the whitespace.

This goes for every modified code-block in this PR.

**Review by StaticRocket — CHANGES_REQUESTED** [review]

File: source/linux/Foundational_Components/Kernel/Kernel_Drivers/QSPI.rst [hunk 3]
```diff
@@ -273,102 +245,106 @@ refer to :file:`Documentation/devicetree/bindings/mtd/jedec,spi-nor.yaml`,
...
-            };
+.. ifconfig:: CONFIG_part_variant in ('AM64X', 'AM62X', 'AM62PX', 'J7200', 'J721E', 'J722S', 'J721S2', 'J784S4','J742S2')
 
-            partition@80000 {
-               label = "ospi.tispl";
-               reg = <0x80000 0x200000>;
-            };
+   The following is an example device-tree node for an OSPI NOR device
+
+   .. code-block:: dts
```
Comment (jsuhaas22) [inline source/linux/Foundational_Components/Kernel/Kernel_Drivers/QSPI.rst:252]: `dts` is not a recognized literal block. I get this warning:

`/home/suhaas/git/processor-sdk-doc/source/linux/Foundational_Components/Kernel/Kernel_Drivers/QSPI.rst:278: WARNING: Could not lex literal_block as "dts". Highlighting skipped.`
Instead you could keep it just as code-block.

**Review by jsuhaas22 — CHANGES_REQUESTED** [review]

**Review by cshilwant — CHANGES_REQUESTED** [review]
Checkout https://github.com/TexasInstruments/processor-sdk-doc/actions/runs/12023717688?pr=50 for errors reported by commit check workflow 

Refer CONTRIBUTING guidelines for more details - https://github.com/TexasInstruments/processor-sdk-doc/blob/master/CONTRIBUTING.md

File: source/linux/Foundational_Components/Kernel/Kernel_Drivers/QSPI.rst (hunk 3 above)
Comment (StaticRocket) [reply source/linux/Foundational_Components/Kernel/Kernel_Drivers/QSPI.rst:252]: Oh no, it is a recognized block. That error just means it couldn't be parsed as written. Incomplete syntax.

https://pygments.org/docs/lexers/#pygments.lexers.devicetree.DevicetreeLexer

**Review by StaticRocket — APPROVED** [review]

**Review by cshilwant — APPROVED** [review]

**Pull request merged**